include/edm4hep/GenToolInfo.h

```diff
@@ -0,0 +1,50 @@
+#ifndef EDM4HEP_GENTOOLINFO_H
+#define EDM4HEP_GENTOOLINFO_H
+
+#include "edm4hep/Constants.h"
+#include "podio/Frame.h"
+#include <string>
+#include <vector>
+
+namespace edm4hep {
+
+struct GenToolInfo {
+  std::string name;
+  std::string version;
+  std::string description;
+
+  GenToolInfo(){};
+  GenToolInfo(const std::string& name, const std::string& version, const std::string& description) :
+      name(name), version(version), description(description){};
+};
+
+namespace utils {
+
+  const std::vector<GenToolInfo> getGenToolInfos(const podio::Frame& frame) {
+    auto toolInfos = std::vector<GenToolInfo>();
+    const auto names = frame.getParameter<std::vector<std::string>>(generatorToolName).value();
+    const auto versions = frame.getParameter<std::vector<std::string>>(generatorToolVersion).value();
+    const auto descriptions = frame.getParameter<std::vector<std::string>>(generatorToolDescription).value();
```

Comment:
In case any of these parameters is not available this implementation will throw a `std::bad_optional_access`, which might be a bit cryptic, but would at least be in line with "fail early in case of problems".

If we wanted a slightly more relaxed approach without exceptions escaping, we could do:
```suggestion
    const auto names = frame.getParameter<std::vector<std::string>>(generatorToolName).value_or({});
    const auto versions = frame.getParameter<std::vector<std::string>>(generatorToolVersion).value_or({});
    const auto descriptions = frame.getParameter<std::vector<std::string>>(generatorToolDescription).value_or({});
```
In that case the worst that can happen is an empty vector, which probably also shows that something went wrong.

I am in favor of the current implementation as that makes errors harder to ignore and easier to diagnose (even if the `std::bad_optional_access` might be a bit cryptic).

Reply:
After some more discussion with @dirkzerwas about how this will be used, I think we can do the "error" handling internally, and simply return an empty vector instead of letting the exception propagate. The main reasons are
- `vector::empty` still let's users check whether information was available
- Information on the tools rather falls into a "nice to have" category, rather than crucially important information for event processing.

Reply:
`value_or` should do then the right thing then